libseccomp/src/error.rs

```diff
@@ -64,6 +64,27 @@ impl Error for SeccompError {
     }
 }
 
+/* Does not work without specialization (RFC 1210) or negative trait bounds
+impl<T: Error> From<T> for SeccompError {
+    fn from(err: T) -> Self {
+        Self::with_source(ErrorKind::Common(err.to_string()), err)
+    }
+}
+*/
+
+macro_rules! impl_seccomperror_from {
+    ($errty:ty) => {
+        impl From<$errty> for SeccompError {
+            fn from(err: $errty) -> Self {
+                Self::with_source(ErrorKind::Common(err.to_string()), err)
```

Comment:
In this case, the output will be `<ErrorType> caused by: <ErrorType>` by [line 51](https://github.com/libseccomp-rs/libseccomp-rs/pull/27/files#diff-0186c07869ec66b16152774dde29b8b526290c7edd8868be341e76072f226bd7R51).
Example of the case of `NulError`:
`NulError  caused by: NulError`

Hence, I'd like to suggest `Errorkind::Common("Convert error", err)` or something like that.
I'm not sure a proper message as generic.

Reply:
The `ToString` implementation of `NulError` (https://doc.rust-lang.org/src/std/ffi/c_str.rs.html#1048-1052) looks like

```rust
impl fmt::Display for NulError {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "nul byte found in provided data at position: {}", self.0)
    }
}
```

so the actually error would look like
`nul byte found in provided data at position: 11 caused by: Err(NulError(11, [72, 101, 108, 108, 111, 32, 87, 111, 114, 108, 100, 0, 33]))`

Is there any reason why we format the source with Debug? If not, what about using Display for the source and adding a new `ErrorKind::Source`?

Reply:
BTW what was the intention of `SeccompError.code`? It is never read. Can we remove it?

Reply:
Yes, but could you leave `code` for `Debug` trait implementation.
I'll add `Debug` soon.